src/webportal/src/app/user/fabric/user-profile/ssh-list-dialog.jsx

```diff
@@ -0,0 +1,128 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import { isEmpty } from 'lodash';
+import {
+  DefaultButton,
+  PrimaryButton,
+  DialogType,
+  Dialog,
+  DialogFooter,
+  TextField,
+} from 'office-ui-fabric-react';
+
+import t from '../../../components/tachyons.scss';
+
+const SSHListDialog = ({ onDismiss, onAddPublickeys }) => {
+  const [error, setError] = useState('');
+  const [inputTitleError, setInputTitleError] = useState('');
+  const [inputValueError, setInputValueError] = useState('');
+  const [processing, setProcessing] = useState(false);
+  const [title, setTitle] = useState('');
+  const [value, setValue] = useState('');
+
+  const onAddAsync = async () => {
+    if (title.trim() === '') {
+      setInputTitleError('Please input title');
+    } else if (value.trim() === '') {
+      setInputValueError('Please input ssk value');
+    } else {
+      setProcessing(true);
+      try {
+        await onAddPublickeys({
+          title: title.trim(),
+          value: value.trim(),
+          time: new Date().getTime(),
+        });
+      } catch (error) {
+        setError(error.message);
+      } finally {
+        setProcessing(false);
+        onDismiss();
+      }
+    }
+  };
+
+  return (
+    <div>
+      <Dialog
+        hidden={false}
+        onDismiss={onDismiss}
+        dialogContentProps={{
+          type: DialogType.normal,
+          title: 'Add SSH Public Keys',
+        }}
+        modalProps={{
+          isBlocking: true,
+        }}
+        minWidth={600}
+      >
+        <div>
+          <div className={t.mt1}>
+            <TextField
+              label='title'
+              required={true}
+              errorMessage={inputTitleError}
+              onChange={e => {
+                setTitle(e.target.value);
+                setInputTitleError(null);
+              }}
+              validateOnFocusOut={true}
+            />
+          </div>
+          <div className={t.mt1}>
+            <TextField
+              label='value'
```

Comment:
Better to give the user some hints.

Value (SSH Public key, starts with `ssh-rsa`):

Reply:
fixed